app/core/pokemon-entity.ts

```diff
@@ -1816,6 +1827,11 @@ export class PokemonEntity extends Schema implements IPokemonEntity {
     ) // prevent resurecting fossils twice
 
     // does not trigger postEffects (iron defense, normal shield, rune protect, focus band, delta orb, flame orb...)
+
+    this.refToBoardPokemon.onSpawn({
```

Comment:
I don't think resurrect() should proc spawn effects. You dont want Sudowoodo to become a tree again,

Reply:
`resurrect()` not proccing spawn effects causes a few bugs. Trees seem to be the only case where it's undesirable to have them trigger again. It's probably best to just use a conditional then.

Reply:
What bugs ? It's the first that came when looking up for onSpawn callbacks. Regigigas -25% attack speed is not desirable neither.

resurrecting doesn't mean spawning a new unit, therefore it should not call onSpawn effects. As simple as that.

Reply:
Another word for resurrect is respawn, hence spawn. It's wholly unintuitive that effects which aren't explicitly stated as triggering "at the start of the fight" would not be reset with everything else on resurrection. Why should Regigigas' passive not be reapplied? Currently its attack speed goes like this: 50 -> 80 -> (resurrect) -> 75 -> 105. So even if you wanted to not reapply its passive, its implementation needs to be changed because this is certainly unintended.

Reply:
Disagree, respawn would imply the entity has been removed and readded, which is not the case for PAC, the entity is always here with a dedicated status. onSpawn is used to specify effects triggering after the entity is created, being at the start of the fight or with a spawning ability like Houndour/Banette. Regigigas passive should not be reapplied, it is designed to be applied once and nothing indicated that it should be reset on resurrection.